app/scripts/modules/amazon/src/instance/details/instance.details.controller.js

```diff
@@ -505,16 +507,69 @@ module(AMAZON_INSTANCE_DETAILS_INSTANCE_DETAILS_CONTROLLER, [
     this.hasHealthState = function hasHealthState(healthProviderType, state) {
       const instance = $scope.instance;
       const healthMetrics = instance.health || [];
-      return healthMetrics.some(function(health) {
+      return healthMetrics.some(function (health) {
         return health.type === healthProviderType && health.state === state;
       });
     };
 
+    const constructInstanceActions = (instance) => {
```

Comment:
My plan is to ultimately have this be in the `AmazonInstanceDetails` component which has any custom methods needed, but is composed of the generic components.